src/Raven.Server/Smuggler/Documents/DatabaseSource.cs

```diff
@@ -70,10 +73,12 @@ public IDisposable Initialize(DatabaseSmugglerOptions options, SmugglerResult re
                 _returnContext = _database.DocumentsStorage.ContextPool.AllocateOperationContext(out _context);
                 _disposeTransaction = _context.OpenReadTransaction();
                 LastEtag = DocumentsStorage.ReadLastEtag(_disposeTransaction.InnerTransaction);
+                LastCompareExchangeIndex = _database.ReadDatabaseRecord().IndexForCompareExchange;
```

Comment:
Should be in next `if`. You need to pass the server transaction to read proper value of this

Reply:
done